enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,585 @@
+	Spec ImagePolicySpec `json:"spec"`
+	// status contains the observed state of the resource.
+	// +optional
+	Status ImagePolicyStatus `json:"status,omitempty"`
+}
+
+// ImagePolicySpec is the specification of the ImagePolicy CRD.
+type ImagePolicySpec struct {
+	// scopes defines the list of image identities assigned to a policy. Each item refers to a scope in a registry implementing the "Docker Registry HTTP API V2".
+	// scopes matching individual images are named Docker references in the fully expanded form, either using a tag or digest. For example, docker.io/library/busybox:latest (not busybox:latest).
+	// More general scopes are prefixes of individual-image scopes, and specify a repository (by omitting the tag or digest), a repository
+	// namespace, or a registry host (by only specifying the host name and possibly a port number) or a wildcarded expression starting with *., for matching all subdomains (not including a port number).
+	// For wildcarded subdomain matching, *.example.com is a valid case, but example*.*.com is not.
+	// For more information about the format, see the document
+	// about the docker transport field:
+	// https://github.com/containers/image/blob/main/docs/containers-policy.json.5.md#docker
+	// +kubebuilder:validation:Required
+	// +required
+	// +listType=set
+	Scopes []ImageScope `json:"scopes"`
+	// policy defines the verification policy for the items in the scopes list
+	// +kubebuilder:validation:Required
+	// +required
+	Policy Policy `json:"policy"`
+}
+
+// ImageScope is the item of the scopes list.
+// +kubebuilder:validation:Pattern=`^\*(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+$|^((((?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+(?::[0-9]+)?)|(localhost(?::[0-9]+)?))(?:(?:/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?)(?::([\w][\w.-]{0,127}))?(?:@([A-Za-z][A-Za-z0-9]*(?:[-_+.][A-Za-z][A-Za-z0-9]*)*[:][[:xdigit:]]{32,}))?$`
+type ImageScope string
+
+// Policy defines the verification policy for the items in the scopes list.
+type Policy struct {
+	// rootOfTrust specifies the public key, the Fulcio certificate and the Rekor public key.
+	// Requires only one of keyData or fulcioCAData must be set.
+	// +required
+	RootOfTrust PolicyRootOfTrust `json:"rootOfTrust"`
+
+	// fulcioSubject specifies OIDC issuer and the email of the fulcio authentication configuration.
+	// Required if rootOfTrust.fulcioCAData is set.
+	// +optional
+	FulcioSubject PolicyFulcioSubject `json:"fulcioSubject,omitempty"`
+
+	// signedIdentity specifies what image identity the signature claims about the image.
+	// +optional
+	SignedIdentity PolicyIdentity `json:"signedIdentity,omitempty"`
+}
+
+// PolicyRootOfTrust defines the public key, the Fulcio certificate and the Rekor public key.
+type PolicyRootOfTrust struct {
+	// keyData contains inline base64 data of the public key.
+	// keyData can be empty if the image got signed keyless.
```

Comment:
What is a keyless signature? I think this could be reworded to be a little clearer
```suggestion
	// May be omitted when the image was signed with a keyless signature.
```

Reply:
> What is a keyless signature?

@saschagrunert could you take a look?

Reply:
So it seems that this keyless is a nickname for using a short lived, identity based token based on OIDC? I think it would be helpful to expand the comments to explain what this actually means

Reply:
To be pedantic about this, there is no such thing as a “keyless signature” — every signature must be with some private key.

At best, it’s “keyless signing”, as in “the user does not _hold/maintain/protect_ a key” (but there is a key with a Fulcio-generated certificate somewhere in there).

---

Maybe we could just delete this sentence instead of dealing with these details? The field documentation already says that `KeyData` and `FulcioCAData` are exclusive alternatives.